### PR TITLE
Cc backend api-design

### PR DESCRIPTION
**Pull request by OpaJaa**
Please comment here what I need to add/change. The folder /api/ contains a generated api.ts. TS/JS -gurus could check that out also to see if these can be used as templates.

### CODE REVIEW

File: cc-backend/openapi.yaml [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+              schema:
+                $ref: "#/components/schemas/Error"
+        "500":
+          description: Internal server error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+
+  /v1/items/{surveyId}:
```
Comment (rvsape) [inline cc-backend/openapi.yaml]: Surveys and items in them are fetched based upon user, so the request parameter would probably be userId

File: cc-backend/openapi.yaml [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+    get:
+      operationId: getItemsFromPK
+      summary: Get all items from PK
+      description: Gets all items from PK via CC-PK-service
```
Comment (rvsape) [inline cc-backend/openapi.yaml]: this request would also get all Items that the user has in the cc database via CC-db-service

File: cc-backend/openapi.yaml [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+  
+  /v1/items/{itemId}:
+    get:
+      operationId: getItemInfoForm
+      summary: Gets item info for form from PK
+      description: Gets item info for Form from PK via CC-PK-service
```
Comment (rvsape) [inline cc-backend/openapi.yaml]: description: also get possible Item configurations and drafts from CC-db-service

File: cc-backend/openapi.yaml [hunk 4]
```diff
@@ -0,0 +1,406 @@
...
+              schema:
+                $ref: "#/components/schemas/Error"
+        "500":
+          description: Internal server error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+
+  /v1/advert/{advertId}:
```
Comment (rvsape) [inline cc-backend/openapi.yaml]: no need for advertId here, since we only get it after succesfull POST to MT

**Review by bunjovis — APPROVED** [review]
Looking good to me. One common comment to everyone doing these API services: be careful when using generated code, you still have to know what the code is doing, how to use it or modify it.

**Review by hannakortetjarvi — APPROVED** [review]

**Pull request merged**